src/core/distributions/phylogenetics/tree/birthdeath/BirthDeathSamplingTreatmentProcess.cpp

```diff
@@ -1590,36 +1594,52 @@ double BirthDeathSamplingTreatmentProcess::simulateDivergenceTime(double origin,
  * \return The diversity (number of species in the reconstructed tree).
  */
 int BirthDeathSamplingTreatmentProcess::survivors(double t) const
-{
-
-    const std::vector<TopologyNode*>& nodes = value->getNodes();
-
+{  
+    
     int survivors = 0;
-    for (std::vector<TopologyNode*>::const_iterator it = nodes.begin(); it != nodes.end(); ++it)
+
+    if ( use_origin )
     {
-        TopologyNode* n = *it;
-        double my_age = n->getAge();
+        if ( t > getOriginAge() )
+        {
+            return 0;
+        }
+        survivors = 1;
+    } else {
+        if ( t > value->getRoot().getAge() )
+        {
+            return 0;
+        }
+        survivors = 2;
+    }
 
-        // my age needs to be smaller that the requested time
-        if ( (my_age - t) < 1E-4 )
+    for (size_t i=0; i<serial_bifurcation_times.size(); ++i) {
+        if (t < serial_bifurcation_times[i])
         {
-            // my parents age needs to be larger/older than the requested time
-            if ( n->isRoot() == false && (n->getParent().getAge() - t) > 1E-4 )
-            {
-                survivors++;
-            }
+            survivors++;
         }
-        else if ( (my_age - t) < -1E-4 )
+    }
+
+    for (size_t i=0; i<serial_tip_ages.size(); ++i) {
+        if (t < serial_tip_ages[i])
         {
-            // my parents age needs to be larger/older than the requested time
-            if ( n->isRoot() == false && (n->getParent().getAge() - t) > -1E-4 )
-            {
-                survivors++;
-            }
+            survivors--;
         }
-        
     }
 
+    for (size_t i=0; i<global_timeline.size(); ++i)
+    {   
+        size_t idx = global_timeline.size() - i - 1;
+        if ( global_timeline[idx] < t ) {
+            break;
+        } else if (global_timeline[idx] > t)
+        {   
+            // TODO by ignoring time = t we implicitly count all tips at a time as survivors
```

Comment:
perhaps remove this todo if it's not a todo anymore

Reply:
```suggestion
            // by ignoring time = t we implicitly count all tips at a time as survivors
```

Yeah, perhaps it's not a to-do anymore but I'd like to leave it as a comment? I don't want to confuse future-me